### PR TITLE
Ports TCC style windows

### PR DESCRIPTION
**Pull request by S34NW**
## What Does This PR Do
Unifies the following windows to match our wall style:

- Glass window
- Reinforced glass window
- Plasma glass window
- Reinforced plasma glass window
- Tinted window

These sprites are likely from skyrat, but are in the TCC style of walls. https://github.com/Skyrat-SS13/Skyrat-tg/pull/1245

Added new smooth groups for regular and reinforced walls, so these windows won't smooth with random crap.

Slightly refactors how window icons are updated

Deletes iced windows, we never used them anyway.

- [x] ~~WIP while I make a system to stop window recolouring affecting the edging~~

## Why It's Good For The Game
Our current windows do not feel like part of the station proper, they are just squares that don't form part of a structure. This changes that to unify everything.

## Images of changes
Before:
ParaCode Test NSS Cyberiad 2022-10-23 135124

After:
ParaCode Test NSS Cyberiad 2022-11-07 165731

Before:
ParaCode Test NSS Cyberiad 2022-10-23 135114

After:
ParaCode Test NSS Cyberiad 2022-11-07 165722

Example with no grilles under the window:
ParaCode Test NSS Cyberiad 2022-11-07 165713

## Testing
Looked at windows, did buildmode stuff

## Changelog
:cl:
tweak: Resprited glass and plasma glass windows to match walls and smooth properly.
/:cl:

### CODE REVIEW

**Comment by OctusGit** [comment]
Based

**Comment by warriorstar-orion** [comment]
I'll be honest, these sort of look like a z-fighting issue, like the game tried to render a window and a wall at the same time. If there were more of a transition between these edges I think they would feel more like integrated parts of the wall.
Untitaled

**Comment by roarbark** [comment]
looks better than current windows. i do agree that if there was a like visual cut wherever it meets walls it would look a lot better, but the sprite as is looks much better than the old windows

File: code/game/objects/structures/window.dm [hunk 1]
```diff
@@ -452,26 +456,39 @@
...
+	var/ratio = obj_integrity / max_integrity
+	ratio = CEILING(ratio * 4, 1) * 25
+	if(ratio <= 75)
+		crack_overlay = mutable_appearance('icons/obj/structures.dmi', "damage[ratio]", -(layer+0.01), appearance_flags = RESET_COLOR)
 		. += crack_overlay
 
+	if(!edge_overlay_file)
+		return
+
+	edge_overlay = mutable_appearance(edge_overlay_file, "[smoothing_junction]", (layer+0.1), appearance_flags = RESET_COLOR)
```
Comment (Edan52) [inline code/game/objects/structures/window.dm:485]:
```suggestion
	edge_overlay = mutable_appearance(edge_overlay_file, "[smoothing_junction]", layer + 0.1, appearance_flags = RESET_COLOR)
```

File: code/game/objects/structures/window.dm [hunk 2]
```diff
@@ -452,26 +456,39 @@
...
+	if(!fulltile)
+		return
 
-		if(ratio > 75)
-			return
-		crack_overlay = mutable_appearance('icons/obj/structures.dmi', "damage[ratio]", -(layer+0.1))
+	var/ratio = obj_integrity / max_integrity
+	ratio = CEILING(ratio * 4, 1) * 25
+	if(ratio <= 75)
+		crack_overlay = mutable_appearance('icons/obj/structures.dmi', "damage[ratio]", -(layer+0.01), appearance_flags = RESET_COLOR)
```
Comment (Edan52) [inline code/game/objects/structures/window.dm:479]:
```suggestion
		crack_overlay = mutable_appearance('icons/obj/structures.dmi', "damage[ratio]", -(layer + 0.01), appearance_flags = RESET_COLOR)
```

**Comment by S34NW** [comment]
> The windows don't even really look like windows, they look more like some weird types of walls, but I guess that's more subjective.

Yeah, that's sorta the point. All windows are parts of a wall, this has not been possible to clearly visually represent with top-down walls. It is however a requirement of 3/4th walls of you don't want them looking like total ass.

**Comment by S34NW** [comment]
> I'll be honest, these sort of look like a z-fighting issue, like the game tried to render a window and a wall at the same time. If there were more of a transition between these edges I think they would feel more like integrated parts of the wall.
> Untitaled
> 

This is a difficult one, as it would need to apply that transition only to borders with walls as opposed to borders with other windows (or you just get the transition repeating along the length of the wall). I will mull over potential solutions.

**Comment by S34NW** [comment]
The changes in https://github.com/ParadiseSS13/Paradise/pull/19598 would go a very long way to improve the contrast concerns with rglass windows next to rwalls

**Comment by AffectedArc07** [comment]
> CI fail is due to a IDB2 bug, and is not indicitive of a broken PR. @AffectedArc07 can confirm that.

pr is fine

File: code/__DEFINES/icon_smoothing.dm [hunk 3]
```diff
@@ -95,7 +95,8 @@ DEFINE_BITFIELD(smoothing_flags, list(
 #define SMOOTH_GROUP_SURVIVAL_TIANIUM_POD S_OBJ(14)		///turf/simulated/wall/mineral/titanium/survival/pod, /obj/machinery/door/airlock/survival_pod, /obj/structure/window/shuttle/survival_pod
 #define SMOOTH_GROUP_HIERO_WALL S_OBJ(15)				///obj/effect/temp_visual/elite_tumor_wall, /obj/effect/temp_visual/hierophant/wall
 #define SMOOTH_GROUP_BRASS_WALL S_OBJ(16)				///turf/simulated/wall/mineral/brass, /obj/structure/falsewall/brass
-
+#define SMOOTH_GROUP_REGULAR_WALLS S_OBJ(17)			///turf/simulated/wall, /obj/structure/falsewall
+#define SMOOTH_GROUP_REINFORCED_WALLS S_OBJ(18)			///turf/simulated/wall/r_wall
```
Comment (hal9000PR) [inline code/__DEFINES/icon_smoothing.dm:99]: doesnt this also contain r_false walls?

Comment (S34NW) [reply code/__DEFINES/icon_smoothing.dm:99]: It needs to, yes. Fix pushed.

Comment (S34NW) [inline code/__DEFINES/icon_smoothing.dm:99]:
```suggestion
#define SMOOTH_GROUP_REINFORCED_WALLS S_OBJ(18)			///turf/simulated/wall/r_wall, /obj/structure/falsewall/reinforced
```

**Comment by S34NW** [comment]
GUH

**Comment by S34NW** [comment]
WIZARD REMAP STEALTH CONFLICT

**Review by hal9000PR — APPROVED** [review]

**Comment by SpringSkipper** [comment]
Not a fan personally. The big dumb square windows do look big and dumb, but this is a tile-based pixel spaceman game. The big dumb windows convey information better, and sometimes that's more important than looking realistic

if you try to make a pixel game look ultra-realistic, you end up with a dull muddy mess.

**Pull request merged**